sstables/storage.cc

```diff
@@ -466,16 +466,15 @@ sstring s3_storage::make_s3_object_name(const sstable& sst, component_type type)
 
 future<> s3_storage::ensure_remote_prefix(const sstable& sst) {
     if (!_remote_prefix) {
-        auto uuid = co_await sst.manager().system_keyspace().sstables_registry_lookup_entry(_location, sst.generation());
-        _remote_prefix = uuid.to_sstring();
+        co_await sst.manager().system_keyspace().sstables_registry_lookup_entry(_location, sst.generation());
+        _remote_prefix = fmt::to_string(sst.generation());
```

Comment:
i think we can go further by removing `ensure_remote_prefix()` or just degrading it by removing
```c++
_remote_prefix = fmt::to_string(sst.generation());
```
so it's more alike an `assert()`. as we can always deduce the "remote_prefix" by formatting `sst.generation()`. BTW, i understand this is a bad name, will change it to `sst.identifier()`.

Reply:
i went ahead, and add this as a separated commit in this PR. as, while testing the tiered storage, i found that we could use `atomic_deleter()` before opening a sstable. in that case, the `_remote_prefix` is not set yet, as we don't call `ensure_remote_prefix()` in `s3_storage::delete_with_system_keyspace()`.